src/coreclr/vm/interpexec.cpp

```diff
@@ -1944,8 +1944,8 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     int32_t vtSize = ip[4];
                     void *vtThis = stack + returnOffset;
 
-                    // clear the valuetype
-                    memset(vtThis, 0, vtSize);
+                    // Interpreter-TODO: Clear the valuetype memory for GC encoding
```

Comment:
@davidwrighton In what scenario is this needed?

Reply:
This depends on how we resolve the GC reporting around calls where the return value overlaps with arguments. We cannot zero the memory unless we ensure that it doesn't overlap with the arguments. 
@kg already has some ideas on that.

Reply:
This should never overlap with arguments. I don't understand why are we removing this zeroing as part of this PR. From what I understand from the spec, newobj needs to zero the new object/vt. I think that ctors fully initializing the object is just a C# requirement.

Reply:
This is encountered at the startup path in `newobj.vt`:
```
Compiled method: .System.RuntimeTypeHandle:GetInstantiationInternal()
Locals size 112
IR_0000: initlocals     [nil <- nil], 16,32
IR_0003: safepoint      [nil <- nil],
IR_0004: ldc.i8.0       [48 <- nil],
IR_0006: mov.8          [16 <- 48],
IR_0009: mov.8          [64 <- 0],
IR_000c: call           [48 <- 64], .System.RuntimeTypeHandle:GetNativeHandle()
IR_0010: mov.vt         [24 <- 48], 8
IR_0014: ldloca         [72 <- nil], 24
IR_0017: newobj.vt      [64 <- 64], .System.Runtime.CompilerServices.QCallTypeHandle:.ctor(byref)
```
The `QCallTypeHandle` is 16 bytes and the offset to the first argument is only 8 bytes. I still don't have a minimal repro, so let's fix this in a follow-up. The change has been reverted.